webapp/channels/src/components/post_emoji/post_emoji.tsx

```diff
@@ -13,23 +13,24 @@ declare module 'react' {
     }
 }
 
-export default class PostEmoji extends React.PureComponent<PostEmojiProps> {
-    public render() {
-        const emojiText = ':' + this.props.name + ':';
+const PostEmoji = ({name, imageUrl}: Props): JSX.Element => {
+    const emojiText = `:${name}:`;
+    const backgroundImageUrl = `url(${imageUrl})`;
 
-        if (!this.props.imageUrl) {
-            return emojiText;
-        }
-
-        return (
-            <span
-                alt={emojiText}
-                className='emoticon'
-                title={emojiText}
-                style={{backgroundImage: 'url(' + this.props.imageUrl + ')'}}
-            >
-                {emojiText}
-            </span>
-        );
+    if (!imageUrl) {
+        return <>{emojiText}</>;
     }
-}
+
+    return (
+        <span
+            alt={emojiText}
+            className='emoticon'
+            title={emojiText}
+            style={{backgroundImage: backgroundImageUrl}}
```

Comment:
(This is not a request change, just a comment on why I would leave it as is)

If we were purist, I would say here something like "this passes to the child component a new object on every render, so we should memoize the value". Nevertheless, since the element is a basic element (`span`) I guess the amount of further re-rendering is negligible. Also, adding the use memo probably will make the component a bit heavier to read.

So... let's leave it like that.